File: docs/guides/kotlin/src/main/kotlin/com/algolia/snippets/searchWithRuleContexts.kt
import com.algolia.client.api.SearchClient
import com.algolia.client.configuration.*
import com.algolia.client.transport.*
import com.algolia.client.extensions.*

import com.algolia.client.model.search.*

val getPlatformTag: () -> String = {
  "" // Implement your logic here
}

suspend fun searchWithRuleContexts() {
  val client = SearchClient(appId = "ALGOLIA_APPLICATION_ID", apiKey = "ALGOLIA_API_KEY")

  val platformTag = getPlatformTag()
  val searchParams = SearchParamsObject(
    query = "<YOUR_SEARCH_QUERY>",
    ruleContexts = listOf(platformTag),
  )

  client.searchSingleIndex(
    indexName = "<YOUR_INDEX_NAME>",
    searchParams = searchParams,
  )
}

File: docs/guides/swift/Sources/searchWithRuleContexts.swift
import Foundation
#if os(Linux) // For linux interop
    import FoundationNetworking
#endif

import Core
import Search

let getPlatformTag = { "" } // Implement your logic here

func searchWithRuleContexts() async throws {
    let client = try SearchClient(appID: "ALGOLIA_APPLICATION_ID", apiKey: "ALGOLIA_API_KEY")

    let platformTag = getPlatformTag()
    let searchParams = SearchSearchParams.searchSearchParamsObject(
        SearchSearchParamsObject(query: "<YOUR_SEARCH_QUERY>", ruleContexts: [platformTag])
    )

    let response: SearchResponse<Hit> = try await client.searchSingleIndex(
        indexName: "<YOUR_INDEX_NAME>",
        searchParams: searchParams
    )
    print(response)
}
